### ISSUE TITLE
Comments received during the SWG approval motion for CQL2

### ISSUE THREAD

**Issue by cportele** [opengeospatial/ogcapi-features#699]
Meeting 2022-06-06: No objections to tagging the current draft of CQL2 as a release candidate were received, but a few comments were raised that we decided to address as follows:

- [x] Move spatial/temporal/array sections from "6.3 Data types and literal values" to the respective requirements classes. @cportele will fix this.
- [x] An `identifier` in double quotes still has to follow the identifier rule. The double quotes are useful to use a property name that is a CQL2 keyword as a property name. Open identifier up to allow all strings in double quotes would be late change and would require a broader discussion first. @pvretano will contact @aaime to see, if the current approach is good for him. If yes, we will leave this as is, otherwise we need to re-discuss.
- [x] The `characterLiteral` currently is broken, in particular `alpha`. It references a deprecated rule. This needs to be fixed and updated. Also, no whitespace is currently allowed in a character literal. @pvretano will fix this.

**Comment by aaime** [comment]
Hi all, answering here. Generally speaking no, having to follow the same identifier rules is limiting, we normally use quoted identifiers when people add space in identifiers, or other special characters. I think we apply the same rules as SQL... for which I cannot find an open specification, but, [quoting from the PostgreSQL documentation](https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS):

> Quoted identifiers can contain any character, except the character with code zero. (To include a double quote, write two double quotes.) This allows constructing table or column names that would otherwise not be possible, such as ones containing spaces or ampersands. The length limitation still applies.

Also as reference, the definition of identifier in our JavaCC grammar:

```
    < #ANY: ~[" ","\""] > |
    < IDENTIFIER: (<LETTER> (<LETTER>|<DIGIT>|<DOLLAR>)*) | <DOUBLE_QUOTE> (<ANY>)+ <DOUBLE_QUOTE> > 
```

**Comment by cportele** [comment]
Meeting 2022-06-20: In addition to the response from @aaime, key RDBMS implementation allow any character between double quotes. We will follow the same approach, @pvretano will include this in the PR for #707.

We can close this issue, the first item was addressed by #708 and #707 covers the rest.
